package/MDAnalysis/coordinates/memory.py

```diff
@@ -0,0 +1,242 @@
+:Copyright: GNU Public License v2
+:Maintainer: Wouter Boomsma <wb@bio.ku.dk>, wouterboomsma on github
+
+
+.. versionadded:: 0.15.0
+
+The module contains a trajectory reader that operates on an array
+in memory, rather than reading from file. This makes it possible to
+use operate on raw coordinate using existing MDAnalysis tools. In
+addition, it allows the user to make changes to the coordinates in
+a trajectory (e.g. through AtomGroup.set_positions) without having
+to write the entire state to file.
+
+
+Examples
+--------
+
+Constructing a Reader from a numpy array
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A simple example where a new universe is created from the
+array extracted from a DCD timeseries
+
+    from MDAnalysis import Universe
+    from MDAnalysisTests.datafiles import DCD, PDB_small
+    from MDAnalysis.coordinates.array import ArrayReader
+
+    universe = Universe(PDB_small, DCD)
+    coordinates = universe.trajectory.timeseries(universe.atoms)
+
+    universe2 = Universe(PDB_small, coordinates,
+                         format=MemoryReader)
+
+This two step process can also be done in one go:
+
+    universe = Universe(PDB_small, DCD, in_memory=True)
+
+"""
+
+import errno
+import numpy as np
+
+from . import base
+
+
+class MemoryReader(base.ProtoReader):
+    """
+    A trajectory reader interface to a numpy array of the coordinates.
+    For compatibility with the timeseries interface, support is provided for
+    specifying the order of columns through the format option.
+
+    Parameter
+    ---------
+    filename : str
+        filename of the trajectory
+    n_atoms : int
+        number of atoms to write
+    convert_units : bool (optional)
+        convert into MDAnalysis units
+    precision : float (optional)
+        set precision of saved trajectory to this number of decimal places.
+    """
+
+    format = 'memory'
+
+    class MemoryTimestep(base.Timestep):
+        """
+        Overrides the positions property in base.Timestep to
+        use avoid duplication of the array.
+        """
+
+        @property
+        def positions(self):
+            return base.Timestep.positions.fget(self)
+
+        @positions.setter
+        def positions(self, new):
+            self.has_positions = True
+            # Use reference to original rather than a copy
+            self._pos = new
+
+    _Timestep = MemoryTimestep
+
+    def __init__(self, coordinate_array, format='afc',
+                 dimensions = None, dt=1, **kwargs):
+        """Constructor
+
+        Parameters
+        ---------
+        coordinate_array : :class:`~numpy.ndarray object
+            The underlying array of coordinates
+        format : str
+            the order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        """
```

Comment:
There is now a `ProtoReader.__init__` to call. Here should be the following line:

```python
super(MemoryReader, self).__init__()
```